### DISCUSSION TITLE
Weather forecast time

### DISCUSSION THREAD

**Discussion by podleyan** [open-meteo/open-meteo#374] [Q&A]
Hello, i would like to test my model on forecast data. I predict electricity demand on next 5 hours every hour. When I look at the data provided by the api i'm not sure from what time forecast is made. 
Is it possible to get data of every hour on next 5 hours? Thank you for your work, it's very useful :)

**Comment by patrick-zippenfenig** [comment]
The API will already return data starting at 0:00 local time. You can use the `time` array and discard any values that are in the past.

You can also set `&forecast_days=2` to limit the API output to 48 hours.

**Reply by podleyan** [reply]
Thanks for your answer, but my question was if it's possible to get from the historical forecast data from when if it was predicted -  1 hour before or whatever. It would be great for me forecast for specific hour - 1-5 hours before, so i'd like to have 5 different values for every hour
Thank you in advance

**Reply by patrick-zippenfenig** [reply]
Do you mean previous weather forecast runs? Like #206?

**Reply by podleyan** [reply]
yes, but for me is enough data for just several hours. i've checked meteoFrance model and if i undestand correctly model runs every 6 hours? so the last over will be time when forecasting - hour +5? or only +1 hour forecasts are stored?

**Reply by patrick-zippenfenig** [reply]
Yes, the MeteoFrance AROME and ARPEGE models are updated every 6 hours. However, other models like the DWD ICON-D2 and ICON-Europe are updated every 3 hours. Some models, such as MetNo or HRRR, are even updated every hour.

I won't keep all the weather model updates for comparison purposes. Instead, I will only store forecasts that are 24, 48, and 72 hours past (maybe even 12 hours, but I haven't decided yet). The differences between consecutive model runs are usually not significant, so what we're interested in is comparing yesterday's forecast to today's forecast. With this information, you can access the forecasts from previous days for any specific hour.
